src/VisualStudio/Core/Def/Implementation/Options/LanguageSettingsPersister.cs

```diff
@@ -220,9 +220,9 @@ private static void SetValueForOption(IOption option, ref LANGPREFERENCES3 langu
         public bool TryFetch(OptionKey optionKey, out object value)
         {
             // This particular serializer is a bit strange, since we have to initially read things out on the UI thread.
-            // Therefore, we refresh the values in the constructor, meaning that this should never get called for our values.
-
-            Contract.ThrowIfTrue(_supportedOptions.Contains(optionKey.Option) && _languageMap.ContainsKey(optionKey.Language));
+            // Therefore, we refresh the values in the constructor, meaning that this should never get called for our values,
+            // unless the constructor fails to fetch the values. We accept that case for now.
```

Comment:
Why did the constructor fail?

Reply:
IDK, but there is this check:

https://github.com/dotnet/roslyn/pull/58675/files#diff-2ba3ebb4519f577240311d8da163bf6e9d0908ea7563065ace41cc93992ed1e7R72

Reply:
Can we add a non-fatal fault report here then so we have a chance to root cause this? The comment there implies we're handling the case where the language isn't installed which seems fine...but if it's not installed then why are we asking for indentation/formatting options for it? That seems suspicious.

Reply:
I don't know if we care about this. The whole LanguageSettingsPersister.cs will go away once we get https://github.com/dotnet/roslyn/issues/56815 implemented.